### ISSUE TITLE
The table is not redrawing when dataSource change in React.StrictMode

### ISSUE THREAD

**Issue by lavoiekeven** [inovua/reactdatagrid#8]
**What edition are you using :** 
Community
**Version for `@inovua/reactdatagrid-community` :** 
4.0.8

**Relevant code or config**
````
import logo from './logo.svg';
import './App.css';

import React from 'react'

import ReactDataGrid from '@inovua/reactdatagrid-community'
import '@inovua/reactdatagrid-community/index.css'
import Button from '@inovua/reactdatagrid-community/packages/Button'

const gridStyle = { minHeight: 550, marginTop: 10 };

const columns = [
  { name: 'id', header: 'Id', defaultVisible: false, type: 'number', maxWidth: 40 },
  { name: 'firstName', header: 'First Name', defaultFlex: 2 },
  { name: 'lastName', header: 'Last Name', defaultFlex: 2 },
  { name: 'email', header: 'Email', defaultFlex: 3 }
];

function App() {
  const [dataSource, setDataSource] = React.useState([]);

  const loadData = () => {
    setDataSource([{ id: 1, firstName: 'test', lastName: "test1", email: 'email@test.com' }]);
  }

  return (
    <div>
      <h3>Sort with remote data</h3>
      <div>
        <Button onClick={loadData}>Load async data</Button>
        <Button
          onClick={() => setDataSource([])}
          style={{ marginLeft: 10 }}
        >Clear data</Button>
      </div>

      <ReactDataGrid
        idProperty="id"
        style={gridStyle}
        columns={columns}
        dataSource={dataSource}
      />
    </div>
  );
}

export default App;
````

**What you did:**
I load data from an array of object. Initialy the array is empty. When I click on the seach button, the array is filled with 1 results.

**What happened:**
The table always shows no record.

**Reproduction repository:**
https://github.com/lavoiekeven/bug-reactdatagird

**Problem description:**
The table is not refreshed when dataSource change

**Suggested solution:**
useEffect trigger on dataSource change ?

**Comment by yurigenin** [comment]
I just tried it and it works fine. Post the code below to any of the source editors on the site:

```
import React from 'react'

import ReactDataGrid from '@inovua/reactdatagrid-enterprise'
import Button from '@inovua/reactdatagrid-community/packages/Button'
import '@inovua/reactdatagrid-enterprise/index.css'

const gridStyle = { minHeight: 550 }

const columns = [
  { name: 'id', header: 'Id', defaultVisible: false, type: 'number', maxWidth: 40 },
  { name: 'firstName', header: 'First Name', defaultFlex: 2 },
  { name: 'lastName', header: 'Last Name', defaultFlex: 2 },
  { name: 'email', header: 'Email', defaultFlex: 3 }
];

function App() {
  const [dataSource, setDataSource] = React.useState([]);

  const loadData = () => {
    setDataSource([{ id: 1, firstName: 'test', lastName: "test1", email: 'email@test.com' }]);
  }

  return (
    <div>
      <h3>Sort with remote data</h3>
      <div>
        <Button onClick={loadData}>Load async data</Button>
        <Button
          onClick={() => setDataSource([])}
          style={{ marginLeft: 10 }}
        >Clear data</Button>
      </div>

      <ReactDataGrid
        idProperty="id"
        style={gridStyle}
        columns={columns}
        dataSource={dataSource}
      />
    </div>
  );
}

export default () => <App />
```

**Comment by lavoiekeven** [comment]
**@inovua/reactdatagrid-community** version not **@inovua/reactdatagrid-enterprise**

I tried with node v12.19.0 and v14.15.0 on Chrome and Edge Chromium.

**Comment by inovua-admin** [comment]
Thanks fort reporting. Strangely enough, the same code is working fine in our examples, though definitely not working in the reproduction repo.
We noticed you're using React 17.0.1 and initially thought that might cause the issue, but even when downgrading to react 16.14.0 it is still reproducible.
We're on it

**Comment by inovua-admin** [comment]
Even more oddly, it does work in the repro repo if you build the app for deployment - `npm run build` and then test in the deployed folder.

**Comment by inovua-admin** [comment]
Turns out it's strict mode - this issue casts more light into the problem https://github.com/facebook/react/issues/15074
We're actively working for coming up with a fix on this - thanks for reporting

**Comment by inovua-admin** [comment]
We've published v `4.0.9` which fixes the issue - you can see our changelog here https://reactdatagrid.io/docs/changelog#4.0.9
